pkg/virtctl/console/console.go

```diff
@@ -58,37 +60,50 @@ func (c *Console) Run(flags *flag.FlagSet) int {
 		return 1
 	}
 
-	interrupt := make(chan os.Signal, 1)
-	signal.Notify(interrupt, os.Interrupt)
+	// Create a round tripper with all necessary kubernetes security details
+	wrappedRoundTripper, err := roundTripperFromConfig(config)
+	if err != nil {
+		log.Println(err)
+		return 1
+	}
 
-	u, err := url.Parse(config.Host)
+	// Create the basic console request
+	req, err := requestFromConfig(config, vm, namespace, device)
 	if err != nil {
-		log.Fatal("dial:", err)
+		log.Println(err)
+		return 1
 	}
-	u.Scheme = "ws"
-	u.Path = fmt.Sprintf("/apis/kubevirt.io/v1alpha1/namespaces/%s/vms/%s/console", namespace, vm)
-	if device != "" {
-		u.RawQuery = "console=" + device
+
+	// Do the call and process the websocket connection with the callback
+	_, err = wrappedRoundTripper.RoundTrip(req)
+
+	if err != nil {
+		log.Println(err)
+		return 1
 	}
-	log.Printf("connecting to %s", u.String())
+	return 0
+}
+
+func WebsocketCallback(ws *websocket.Conn, resp *http.Response, err error) error {
 
-	ws, resp, err := websocket.DefaultDialer.Dial(u.String(), nil)
 	if err != nil {
 		if resp != nil && resp.StatusCode != http.StatusOK {
 			buf := new(bytes.Buffer)
 			buf.ReadFrom(resp.Body)
-			log.Fatalf("Can't connect to console (%d): %s\n", resp.StatusCode, buf.String())
+			return fmt.Errorf("Can't connect to console (%d): %s\n", resp.StatusCode, buf.String())
```

Comment:
Kuberenetes convention is to always use lower case error messages (because they generally get chained together). Is that a convention we follow?